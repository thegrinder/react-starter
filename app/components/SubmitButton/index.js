import React from 'react';
import { Button } from 'basic-styled-uikit';
import Spinner from '../Spinner';

const SubmitButton = props => (
  <Button type="submit" renderSpinner={<Spinner color="white" />} {...props} />
);

export default SubmitButton;
