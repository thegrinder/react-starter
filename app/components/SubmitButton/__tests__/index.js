import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import theme from '../../../theme/components';
import SubmitButton from '..';


const renderComponent = () => render(
  <ThemeProvider theme={theme}>
    <SubmitButton>click me</SubmitButton>
  </ThemeProvider>,
);

describe('<SubmitButton />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { container: { firstChild } } = renderComponent();
    expect(firstChild).toBeDefined();
    expect(firstChild.type).toEqual('submit');
    expect(firstChild).toMatchSnapshot();
  });
});
