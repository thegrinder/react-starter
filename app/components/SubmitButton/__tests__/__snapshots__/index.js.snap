// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> should render correctly 1`] = `
<button
  class="StyledButton-k0ad56-0 hzwlwW"
  type="submit"
>
  <span
    class="ChildrenWrapper-sc-30uoyc-0 cdOdhF"
  >
    click me
  </span>
</button>
`;
